OptimizelySDK.Tests/OptimizelyUserContextTest.cs

```diff
@@ -443,6 +504,32 @@ public void DecideForKeysWithOneFlag()
             Assert.IsTrue(TestData.CompareObjects(decision, expDecision));
         }
 
+        #endregion DecideForKeys
+
+        #region DecideAll
+
+        [Test]
+        public void DecideAllWithUspShouldOnlyLookupSaveOnce()
+        {
+            var userProfileServiceMock = makeUserProfileServiceMock();
+            var optimizely = new Optimizely(TestData.Datafile, EventDispatcherMock.Object,
+                LoggerMock.Object, ErrorHandlerMock.Object, userProfileServiceMock.Object);
+            var user = optimizely.CreateUserContext(UserID);
+
+            _ = user.DecideAll();
+
+            LoggerMock.Verify(
+                l => l.Log(LogLevel.INFO,
+                    "We were unable to get a user profile map from the UserProfileService."),
+                Times.Never);
+            LoggerMock.Verify(
+                l => l.Log(LogLevel.ERROR, "The UserProfileService returned an invalid map."),
+                Times.Never);
+            userProfileServiceMock.Verify(l => l.Lookup(UserID), Times.Once);
+            userProfileServiceMock.Verify(l => l.Save(It.IsAny<Dictionary<string, object>>()),
+                Times.Once);
+        }
+
```

Comment:
These tests look good. Wondering if we have tests covering the contents of the final UPS save call, combining all decisions in the session.

Reply:
Can we also have the similar coverage (lookup once - save once, the ups save contents) with legacy APIs (activate, getVariation) for regression?

Reply:
Added coverage of the final UPS `Save` as requested.  

I'm researching the `Activate` and `GetVariation` tests where UPS is used.